PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/configuration/ui/AzureFunctionsConfigurationPanel.kt

```diff
@@ -58,16 +56,19 @@ import javax.swing.table.TableCellEditor
 import javax.swing.table.TableCellRenderer
 
 @Suppress("UnstableApiUsage")
-class AzureFunctionsConfigurationPanel: AzureRiderAbstractConfigurablePanel {
+class AzureFunctionsConfigurationPanel : AzureRiderAbstractConfigurablePanel, Disposable {
 
     private val properties: PropertiesComponent = PropertiesComponent.getInstance()
 
     private lateinit var coreToolsEditorModel: ListTableModel<AzureRiderSettings.AzureCoreToolsPathEntry>
     private lateinit var coreToolsEditor: JBTable
-    private lateinit var coreToolsDownloadPathEditor: TextFieldWithBrowseButton
 
     private val isCoreToolsFeedEnabled = Registry.`is`("azure.function_app.core_tools.feed.enabled")
 
+    init {
+        Disposer.register(application, this)
+    }
+
```

Comment:
Note that using an `application` as `Disposable` is mostly a no-op: you won't observe termination of such `Disposable` in any normal scenarios.

It would be much better to accept a parent disposable in constructor, or call `Disposer.register` in a place where you create this panel.

`AzureRiderConfigurable` should already be disposable, and if not, then you can make it so and the IntelliJ container will take care of its disposal. So, it would be a more appropriate parent for this type.

Reply:
Thanks for the hint, that works indeed! Have fixed this in #652: https://github.com/JetBrains/azure-tools-for-intellij/pull/652/commits/cf68fcb6e84651577db8c93982a6e91ee912d3de

If OK, can you accept changes in this PR as well? I'll then merge both in.

Reply:
Reviewed those changes and this is okay.